DuckDuckGo/Onboarding/ContextualOnboarding/TrackerMessageProvider.swift

```diff
@@ -0,0 +1,185 @@
+//
+//  TrackerMessageProvider.swift
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import PrivacyDashboard
+import TrackerRadarKit
+import BrowserServicesKit
+
+protocol TrackerMessageProviding {
+    func trackersType(privacyInfo: PrivacyInfo?) -> OnboardingTrackersType?
+    func trackerMessage(privacyInfo: PrivacyInfo?) -> NSAttributedString?
+}
+
+struct MajorTrackers {
+    static let facebookDomain = "facebook.com"
+    static let googleDomain = "google.com"
+
+    static let domains = [facebookDomain, googleDomain]
+}
+
+enum OnboardingTrackersType: Equatable {
+    case majorTracker
+    case ownedByMajorTracker(owner: Entity)
+    case blockedTrackers(entityNames: [String])
+    case noTrackers
+}
+
+struct TrackerMessageProvider: TrackerMessageProviding {
```

Comment:
Not for now but in the future we can probably move this and share it between iOS and macOS

Reply:
Yeah it will need some reworking to work with both but would definitely be good… Maybe we should make a list of cleanup/nice to do task (including extracting some of the generic UI etc from the onboarding module.